### DISCUSSION TITLE
issues while using TriggerDagRunOperator in airflow 2.6.3

### DISCUSSION THREAD

**Discussion by Amanwa19** [apache/airflow#35976] [Q&A]
Hi All,

Need your expertise with the issue . 
Airflow Version - 2.6.3
Python - 3.8
Amazon providers - 8.10.0 

We are getting below error while using TriggerDagRunOperator -

File "/home/afadmin/.local/lib/python3.8/site-packages/packaging/utils.py", line 34, in canonicalize_name value = _canonicalize_regex.sub("-", name).lower() TypeError: expected string or bytes-like object

snippet : 
next_task = TriggerDagRunOperator(
    task_id='next_task',
    trigger_dag_id='Child_dag'
    #python_callable=m
    )
 

Thanks

**Comment by potiuk** [comment]
I am guessing you are using poetry https://github.com/python-poetry/poetry/issues/2028 this is not supported nor recommended by Airflow - using `pip` and `constratints` is the only way Airflow should be instlled in reproducible way. 

See notes here:

https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html 

Alternatively you can download the constraints and manually see what are the different versions of packages you got installed and manually update them in your poetry configuration (and keep it updated for future version).

Or you can vote and advocate for Poetry implementing constraint mechanism https://github.com/python-poetry/poetry/issues/7051

**Reply by Amanwa19** [reply]
Thanks for the reply @potiuk . I have used pip and constratints only for Airflow Installation.
hostname: /home/af/airflow/> poetry --version
-bash: poetry: command not found
We had recently upgraded the Amazon provider from 7.4.1 to 8.10.0, and it seems afterwards we started facing the issue .
Thinking about upgrading providers to 8.11.0 ..!!

Any suggestions will be appreciated,  Thank you 😊

**Reply by potiuk** [reply]
Maybe if you show the whole stack-trace that would be easier to guess - but likely it's some compatibility issue of some package that we are not aware of - but what package and whet compatibility it's hard to say without seeing the whole stacktrace - the single line is not enouhg to guess (that's why I guessed poetry because it happened for others with poetry).

**Reply by Amanwa19** [reply]
Thanks @potiuk , upgrading Amazon provider to 8.11.0 seems to have resolved the issue.
I will share all the relevant details here .
Thanks

**Reply by potiuk** [reply]
cool
